3scale/porta
# THREESCALE-11500: Fix confusing design flaw in "promote" UI (Part 1)

[THREESCALE-11500: Fix confusing design flaw in "promote" UI](https://issues.redhat.com/browse/THREESCALE-11500)

This issue exposes two problems, this PR deals with the first part: 
"The Promote button is not enabled immediately when changes are done. Instead a job is enqueued and the button is enabled when this job is executed."

With the help of @jlledom, we've reached to the conclusion that not knowing when a job is going to be executed is a problem here. In this PR, we're removing the job who deals with updating `proxy_configs_affecting_changes` table and, instead, we propose to update the table directly at the `ProxyConfigEventSubscriber`. We don't think this will imply any substantial downgrade in the performance but we're open to discussion.

**Verification steps** 

1. Go to Products > Integrations
2. Make a change there, could be adding a Mapping Rule or modifying the Policies, for example
3. Go to Integrations > Configuration and confirm the Promote button is enabled
